### ISSUE TITLE
Audio element 'label' attribute not working?

### ISSUE THREAD

**Issue by BrechtDeMan** [BrechtDeMan/WebAudioEvaluationTool#187]
In an AB test, I thought you should be able to name certain elements e.g. custom references/anchors or a non-blind demo (which I'm currently making). Is that not the way a 'label' works, or is it just a bug that it doesn't work here? 

The wiki currently says
> `label`: string, optional. Use this if you want to specify how this sample is labelled, e.g. when it is a reference you can label the reference button 'Example' or 'Click to hear the original'.

See for instance this XML specification: 
`<waet xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="test-schema.xsd"><setup interface="AB"><exitText>Thank you.</exitText><metric><metricenable>testTimer</metricenable><metricenable>elementTimer</metricenable><metricenable>elementTracker</metricenable><metricenable>elementFlagListenedTo</metricenable><metricenable>elementListenTracker</metricenable></metric><interface><interfaceoption type="show" name="volume"></interfaceoption><interfaceoption type="show" name="playhead"></interfaceoption></interface></setup><page id="keamspage" hostURL="test-data/keams/" loop="true" synchronous="true"><commentboxprefix>Comment on track</commentboxprefix><interface></interface><audioelement url="auto.wav" id="auto" gain="0" label="Legacy" name="legacy"></audioelement><audioelement url="keams.wav" id="keams" gain="0" label="Knowledge-engineered" name="knowledge-engineered"></audioelement><audioelement url="pro.wav" id="pro" gain="0" label="Pro human" name="pro-human"></audioelement></page></waet>`

**Comment by BrechtDeMan** [comment]
Same problem for MUSHRA: `label` doesn't do anything

**Comment by nickjillings** [comment]
This should be fixed for AB testing. Will need to edit some code in all the others to support custom labels

**Comment by BrechtDeMan** [comment]
Thanks!
Confirmed that it works. 

Perhaps a sensible property would be auto-scaling of text when at least one string is too long? See attached - I could imagine typical names being at least this long. 
Don't think this needs a separate parameter, best to keep things simple. 
Also note that the 'overflow' happens only on the right of the box. 
For now it's great though. 

![ab-labels](https://cloud.githubusercontent.com/assets/4103243/24012900/f9d04884-0a7f-11e7-83c0-53079496d81f.png)

**Comment by nickjillings** [comment]
It gets tricky. What if this were on the APE!

But AB could be better. Should be improved by replacing the - with a space instead

Sent from my iPad

On 16 Mar 2017, at 18:39, Brecht De Man <notifications@github.com<mailto:notifications@github.com>> wrote:


Thanks!
Confirmed that it works.

Perhaps a sensible property would be auto-scaling of text when at least one string is too long? See attached - I could imagine typical names being at least this long.
Don't think this needs a separate parameter, best to keep things simple.
Also note that the 'overflow' happens only on the right of the box.
For now it's great though.

[ab-labels]<https://cloud.githubusercontent.com/assets/4103243/24012900/f9d04884-0a7f-11e7-83c0-53079496d81f.png>

**Comment by BrechtDeMan** [comment]
You're right. Not sure what to do with APE. Perhaps only showing the full string while it's playing, or being moved... 
Either way this would be very low priority, and one should probably just use single characters or very short strings as labels anyway. 

Hyphen doesn't really do anything in this case except, you know, removing the hyphen. But K.E. works!

**Comment by nickjillings** [comment]
Is there anything further for this one?

**Comment by BrechtDeMan** [comment]
If all interfaces have some custom label functionality (even if it's single letter labels like on APE, perhaps you'd like to do 'S' for synthesised model, 'R' for recording, 'V' for vocalisation, ...) then this can be closed. 

Of course usually you would want 123... or abc... labels randomly assigned, but sometimes you just want to audition some sounds knowing which is which.

**Comment by djmoffat** [comment]
I have committed update to fix this label issue for discrete and likert scales
